### PR TITLE
Adds Rosetta in case the laptop has a M1 chip

### PR DESCRIPTION
**Pull request by laicuRoot**
This PR introduces a change to the script to check if a Mac computer is running on the M1 architecture. Common tools like AWSCLI or MongoDB require Rosetta to be installed in order to run on M1 Macs.

Tested in M1 virtual machine. 
**Before:** 
image
**After:**
image

image

- [x] Updated CHANGELOG
- [x] Updated README.md (if necessary)

### CODE REVIEW

**Comment by cpytel** [comment]
Thanks, @laicuRoot. When you say "some packages" what do you mean? Are they packages installed by the Laptop script itself? I test the Laptop script on a fresh install of macOS on M1 which does not have Rosetta installed, so it should not be a dependency of the script itself.

If your talking about some future packages an individual developer might install, is that truly common enough to proactively install it on every new system, or is it better to only install it when actually needed for something you're working on?

**Comment by laicuRoot** [comment]
> Thanks, @laicuRoot. When you say "some packages" what do you mean? Are they packages installed by the Laptop script itself? I test the Laptop script on a fresh install of macOS on M1 which does not have Rosetta installed, so it should not be a dependency of the script itself.
> 
Thank you for the comment @cpytel. Yes you are right, the script itself works perfectly, I have poorly worded the description of the PR. With some packages I mean tools like `AWSCLI` or such as `MongoDB`. I considered adding this requirement to the script after encountering an error during app setup. The error message was not very informative, so I had to conduct some research.
After that I thought about adding a note to the Readme but then, after discussion, decided that it would be better to have the laptop script checking if Rosetta is needed as it is a common installation that one would need to do rather than adding a note in every repository where AWSCLI is needed. 

> If your talking about some future packages an individual developer might install, is that truly common enough to proactively install it on every new system, or is it better to only install it when actually needed for something you're working on?
>
In my opinion I think is quite common nowadays but I also agree that there might be cases where is not needed at all. I was thinking that perhaps a good idea would be to prompt the user to confirm if they want to install it or not. Something like: 
```
[...code checking arch above]
   echo "Do you want to install Rosetta? (y/n)"
        read answer
        if [[ "$answer" == "y" ]]
        then
          # Install Rosetta2
          /usr/sbin/softwareupdate --install-rosetta --agree-to-license
        else
          echo "Installation cancelled. Exiting script."
          exit 0
        fi
```
What do you think about this?

**Comment by cpytel** [comment]
We don't have any other options on the script, generally this is intended to be an opinion and straightforward setup that works for everyone.

Under that philosophy, we want to make the choices for people about what is necessary or the best setup. If someone wants to go against that default, then they can modify the script to suite themselves.

I think Rosetta is probably OK to add as a default now, but we should monitor it in the future and remove it when its no longer necessary for most people.

**Comment by laicuRoot** [comment]
> We don't have any other options on the script, generally this is intended to be an opinion and straightforward setup that works for everyone.
> 
> Under that philosophy, we want to make the choices for people about what is necessary or the best setup. If someone wants to go against that default, then they can modify the script to suite themselves.

Thank you for the comment @cpytel

> 
> I think Rosetta is probably OK to add as a default now, but we should monitor it in the future and remove it when its no longer necessary for most people.

Great! I will refactor the code and set it up ready for review 🙂 . 
Definitely will keep an eye out in case we need to remove it, thanks!

File: mac [hunk 1]
```diff
@@ -79,6 +79,22 @@ case "$SHELL" in
     ;;
 esac
 
+# checks architecture
+if [[ $(uname -m) == "arm64" ]]
+  then
+  echo "This Macbook has an M1 architecture."
```
Comment (mike-burns) [inline mac:85]: I don't feel strongly (don't have an M1) but this feels a bit noisy. Up to you whether you think otherwise.

Comment (laicuRoot) [reply mac:85]: You are right, the cleaner the better.

**Pull request merged**